working_villagers/jobs/hider_v1.lua

```diff
@@ -0,0 +1,656 @@
+  if #path == 0   then return false, nil end
+  if trace_can_traverse then print('hider.can_traverse() 11') end
+--print('path found')
+  return true, path
+end
+
+function hider.is_player_near(v, pos)
+  --print('hider.is_player_near() 1')
+  assert(v   ~= nil)
+  --print('hider.is_player_near() 2')
+  assert(pos ~= nil)
+  --print('hider.is_player_near() 3')
+  local player,player_position = v:get_nearest_player(search_radius,pos)
+  --print('hider.is_player_near() 4')
+  if player          == nil then return false end
+  --print('hider.is_player_near() 5')
+  if player_position == nil then return false end
+  --print('hider.is_player_near() 6')
+  return true
+end
+
+function hider.can_fit(pos)
+  --print('hider.can_fit() 1')
+  assert(pos ~= nil)
+  --print('hider.can_fit() 2')
+  local up
+  local air
+  
+  if not func.walkable_pos(pos) then return false end
+
+  up  = vector.add(pos, up_vec1)
+  if not func.walkable_pos(up) then return false end
+--  --print('hider.can_fit() 3')
+--  air = minetest.get_node(up)
+--  --print('hider.can_fit() 4')
+--  assert(air.name=="air")
+--  --print('hider.can_fit() 5')
+--
+  up  = vector.add(pos, up_vec2)
+  if not func.walkable_pos(up) then return false end
+--  --print('hider.can_fit() 6')
+--  air = minetest.get_node(up)
+--  --print('hider.can_fit() 7')
+--  if air.name ~= "air" then return false end
+--  --print('hider.can_fit() 8')
+
+  return true
+end
+
+function filter(pred, list)
+  --print('filter() 1')
+  assert(pred ~= nil)
+  --print('filter() 2')
+  assert(list ~= nil)
+  --print('filter() 3')
+  local result = {}
+  --print('filter() 4')
+  for _, pos in ipairs(list) do
+    --print('filter() 5')
+    local flag = pred(pos)
+    if flag == true then
+      --print('filter() 6')
+      table.insert(result, pos)
+      --print('filter() 7')
+    else assert(flag == false)
+    end
+    --print('filter() 8')
+  end
+  --print('filter() 9')
+  assert(#result <= #list)
+  --print('filter() 10')
+  return result
+end
+function vfilter(v, pred, list)
+  --print('filter() 1')
+  assert(pred ~= nil)
+  --print('filter() 2')
+  assert(list ~= nil)
+  --print('filter() 3')
+  local result = {}
+  --print('filter() 4')
+  for _, pos in ipairs(list) do
+    --print('filter() 5')
+    local flag = pred(v, pos)
+    if flag == true then
+      --print('filter() 6')
+      table.insert(result, pos)
+      --print('filter() 7')
+    else assert(flag == false)
+    end
+    --print('filter() 8')
+  end
+  --print('filter() 9')
+  assert(#result <= #list)
+  --print('filter() 10')
+  return result
+end
+
+--function partial(f, head)
+--  --print('partial() 1')
+--  assert(f    ~= nil)
+--  --print('partial() 2')
+--  assert(head ~= nil)
+--  --print('partial() 3')
+--  return function(...)
+--    f(head, unpack(...))
+--  end
+--end
+
+local trace_find_path = false
+function find_path(v, list)
+  if trace_find_path then print('hider.find_path() 1') end
```

Comment:
It would be nice if the debugging could be removed from any features before they are added to the main branch, if this is still in development, then exclude it from the pull request please.

Reply:
I need to learn more fine-grained logging for lua, because this awful idiom is popping up all over my .minetest/mods sub tree